aten/src/ATen/native/Correlation.cpp

```diff
@@ -0,0 +1,70 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+namespace native {
+
+Tensor cov(
+    const Tensor& self,
+    int64_t correction,
+    const c10::optional<Tensor>& fweights,
+    const c10::optional<Tensor>& aweights) {
+  TORCH_CHECK(
+      self.ndimension() <= 2, "cov(): input has more than 2 dimensions");
+
+  // View input tensor as 2D (variables, observations)
+  auto in = self.ndimension() < 2 ? self.view({1, -1}) : self;
+  const auto num_observations = in.size(1);
+
+  Tensor w;
+
+  if (fweights.has_value()) {
+    const auto& fw = fweights.value();
+    TORCH_CHECK(
+        fw.ndimension() <= 1, "cov(): fweights has more than 1 dimension");
+    TORCH_CHECK(
+        at::isIntegralType(fw.scalar_type(), false),
+        "cov(): fweights must be integral dtype");
+    TORCH_CHECK(
+        fw.numel() == num_observations,
+        "cov(): incompatible number of observations and fweights");
```

Comment:
we should be clearer about what is "compatible" -- the expectation is that fweights has the same numel as there are observations (rows) of input